### PR TITLE
Add associated pull requests and commit compare functionality 

### PR DESCRIPTION
**Pull request by matthewgapp**
Adds API support for 

- [associated pull requests](https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#list-pull-requests-associated-with-a-commit)
- [compare commits](https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#compare-two-commits)

### CODE REVIEW

File: src/models/commits.rs
Comment (matthewgapp) [inline src/models/commits.rs:1]: These types were created automatically using the json schema [here](https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#compare-two-commits)

The types were generated https://quicktype.io/

File: src/models/commits.rs [hunk 1]
```diff
@@ -22,3 +22,129 @@ pub struct Comment {
...
+    // unlike the schema online, this can be null
+    pub blob_url: Option<String>,
+    pub changes: i64,
+    pub contents_url: String,
+    pub deletions: i64,
+    pub filename: String,
+    pub patch: Option<String>,
+    pub previous_filename: Option<String>,
+    // unlike the schema online, this can be null
+    pub raw_url: Option<String>,
```
Comment (matthewgapp) [inline src/models/commits.rs:107]: discovered this while using the endpoints

File: src/api/commits/compare_commit.rs [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    }
+
+    /// Page number of the results to fetch.
+    pub fn page(mut self, page: impl Into<u32>) -> Self {
+        self.page = Some(page.into());
+        self
+    }
+
+    /// Sends the actual request.
+    pub async fn send(self) -> crate::Result<models::commits::CommitComparison> {
```
Comment (matthewgapp) [inline src/api/commits/compare_commit.rs:43]: this endpoint accepts pagination params but doesn't return an array so I didn't wrap the return type in `Page`

File: examples/list_gists_for_token_holder.rs [hunk 3]
```diff
@@ -29,7 +29,7 @@ async fn main() -> octocrab::Result<()> {
         println!(
             "{id} | {url} | [{files}] | {description}",
             id = gist.id,
-            url = gist.html_url.to_string(),
+            url = gist.html_url,
```
Comment (matthewgapp) [inline examples/list_gists_for_token_holder.rs:32]: clippy --fix

File: examples/create_a_gist.rs [hunk 4]
```diff
@@ -19,6 +19,6 @@ async fn main() -> octocrab::Result<()> {
         .public(false)
         .send()
         .await?;
-    println!("Done, created: {url}", url = gist.html_url.to_string());
+    println!("Done, created: {url}", url = gist.html_url);
```
Comment (matthewgapp) [inline examples/create_a_gist.rs:22]: clippy --fix

File: tests/current_user_orgs_test.rs [hunk 5]
```diff
@@ -18,13 +18,13 @@ async fn setup_api(template: ResponseTemplate) -> MockServer {
     let mock_server = MockServer::start().await;
 
     Mock::given(method("GET"))
-        .and(path(format!("/user/memberships/orgs")))
+        .and(path("/user/memberships/orgs".to_string()))
```
Comment (matthewgapp) [inline tests/current_user_orgs_test.rs:21]: clippy --fix

File: tests/current_user_orgs_test.rs [hunk 6]
```diff
@@ -18,13 +18,13 @@ async fn setup_api(template: ResponseTemplate) -> MockServer {
...
         .respond_with(template)
         .mount(&mock_server)
         .await;
     setup_error_handler(
         &mock_server,
-        &format!("GET on /user/membership/orgs was not received"),
+        &"GET on /user/membership/orgs was not received".to_string(),
```
Comment (matthewgapp) [inline tests/current_user_orgs_test.rs:27]: clippy --fix

File: src/models/events/payload/workflow_run.rs [hunk 7]
```diff
@@ -20,6 +16,7 @@ pub struct WorkflowRunEventPayload {
 #[non_exhaustive]
 pub enum WorkflowRunEventAction {
     Requested,
+    InProgress,
```
Comment (matthewgapp) [inline src/models/events/payload/workflow_run.rs:19]: flyby fix

**Comment by XAMPPRocky** [comment]
Thank you for your PR! Since I merged your other PR, you'll need to rebase these changes to remove the similar changes. In the future I would recommend making the things like clippy --fix a separate PR, as then I could merge all three PRs at once.

**Comment by XAMPPRocky** [comment]
> I think the "Squash and merge" strategy will take care of dropping the duplicate changes.

To be clear, it's also about review, it's harder for me to review a PR if the same changes exist.

**Comment by matthewgapp** [comment]
Thanks for the review and apologies for the delay. Rebased.

**Comment by XAMPPRocky** [comment]
@matthewgapp You need to update your tests to `#[tokio::test]`, they're currently failing for not being in a Tokyo runtime.

**Comment by matthewgapp** [comment]
> @matthewgapp You need to update your tests to `#[tokio::test]`, they're currently failing for not being in a Tokyo runtime.

Apologies. I should have ran cargo test. I didn't see any signs that `crate::Octocrab::default()` needed an async runtime but I guess internally it spawns a tokio task when it creates a tower middleware buffer service.

**Comment by XAMPPRocky** [comment]
Thank you for your PR!

**Pull request merged**